Modules/core/Controller/CorespaceaccessController.php

```diff
@@ -349,4 +343,59 @@ public function pendingusereditAction($id_space, $id) {
         ));
     }
 
+    /**
+     * 
+     * Reject a pending user
+     * (sets validate=0 && validated_by=<logged user id> in core_pending_accounts)
+     * 
+     * @param int $id_space
+     * @param int $id_pendingAccount
+     */
+    public function pendinguserdeleteAction($id_space, $id_pendingAccount) {
+        $this->checkAuthorization(CoreStatus::$ADMIN);
+        $modelPending = new CorePendingAccount();
+        $modelPending->invalidate($id_pendingAccount, $_SESSION["id_user"]);
+        $mailParams = ["id_space" => $id_space, "id_user" => $modelPending->get($id_pendingAccount)["id_user"]];
+        $this->notifyUserByEmail($mailParams, "reject_pending_user");
+        $this->redirect("corespacependingusers/".$id_space);
+    }
+
+    /**
+     * 
+     * Send an Email to user. Manage the following cases :
+     * - user account has been created
+     * - user is accepted as a member of space
+     * - user request to join a space is rejected
+     *
+     * @param array $params required to fill sendEmail() parameters. Depends on why we want to notify user
+     * @param string $origin determines how to get sendEmail() paramters from $params
+     */
+    public function notifyUserByEmail($params, $origin) {
+        Configuration::getLogger()->debug("notifyUserByEmail", ["params" => $params]);
+        $lang = $this->getLanguage();
+        $fromName = "Platform-Manager";
+        $from = Configuration::get('smtp_from');
+        $spaceName = $params["space_name"];
```

Comment:
not always set, in function pendinguserdeleteAction for example
so:
* all functions should set space_name if possible (not superadmin space for exemple)
* here, if space_name is not set, then should not be set in subject